src/tools/rbd_mirror/LeaderWatcher.h

```diff
@@ -98,6 +98,8 @@ class LeaderWatcher : protected librbd::Watcher {
         watcher(watcher) {
     }
 
+    virtual ~LeaderLock() = default;
```

Comment:
Think you wanted to put that in the mocked back class [1]

[1] https://github.com/adamemerson/ceph/blob/1dc449ac5d8ed087af677b61d6da94113bbdf7e4/src/test/rbd_mirror/test_mock_LeaderWatcher.cc#L31